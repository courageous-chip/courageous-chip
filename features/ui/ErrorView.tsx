import { useTheme } from "@react-navigation/native";
import React, { FC } from "react";
import { StyleSheet, Text, View } from "react-native";

export const ErrorView: FC = function () {
  const {
    colors: { background: backgroundColor },
  } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor }]}>
      <Text style={styles.text}>ðŸ’©</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
  text: {
    fontSize: 72,
  },
});
