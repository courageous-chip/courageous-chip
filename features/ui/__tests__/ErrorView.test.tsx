import React from "react";
import renderer from "react-test-renderer";

import { ErrorView } from "../ErrorView";

describe("<ErrorView />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ErrorView />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
