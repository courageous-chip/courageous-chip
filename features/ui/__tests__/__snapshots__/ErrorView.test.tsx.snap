// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorView /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(242, 242, 242)",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 72,
      }
    }
  >
    ðŸ’©
  </Text>
</View>
`;
